### PR TITLE
[bgp] Improve test_bgp_bbr

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Summary:
Fixes https://github.com/Azure/sonic-buildimage/issues/8425

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Improve the robustness of `test_bgp_bbr`

#### How did you do it?
Improve test_bgp_bbr:
1. for the route check on the first ToR that receives the update and
DUT, let's keep checking for a period of time to see if the route's
presence.
2. for the route check on the DUT, add extra check to verify the route
is advertised to the selected VMs.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

#### How did you verify/test it?
Run over `t1` and `t1-backend`

### CODE REVIEW

File: tests/bgp/test_bgp_bbr.py [hunk 1]
```diff
@@ -210,32 +212,49 @@ def announce_routes(routes):
...
+            if not dut_route_aspath == dut_route_aspath_expected:
+                logging.warn('On DUT expected aspath: {}, actual aspath: {}'.format(dut_route_aspath_expected, dut_route_aspath))
+                return False
+            if 'advertisedTo' not in dut_route:
+                logging.warn("DUT doesn't advertise the route")
+                return False
+            advertised_to = set([bgp_neighbors[_]['name'] for _ in dut_route['advertisedTo']])
+            for vm in other_vms:
+                if vm not in advertised_to:
+                    logging.warn("DUT doesn't advertise route to neighbor %s" % vm)
```
Comment (yxieca) [inline tests/bgp/test_bgp_bbr.py:252]: Really minor: should be "didn't" instead of "doesn't"

**Review by yxieca — APPROVED** [review]

File: tests/bgp/test_bgp_bbr.py (hunk 1 above)
Comment (lolyu) [reply tests/bgp/test_bgp_bbr.py:252]: Will update, thanks!

File: tests/bgp/test_bgp_bbr.py [hunk 2]
```diff
@@ -210,32 +212,49 @@ def announce_routes(routes):
...
+            if not dut_route_aspath == dut_route_aspath_expected:
+                logging.warn('On DUT expected aspath: {}, actual aspath: {}'.format(dut_route_aspath_expected, dut_route_aspath))
+                return False
+            if 'advertisedTo' not in dut_route:
+                logging.warn("DUT didn't advertise the route")
+                return False
+            advertised_to = set([bgp_neighbors[_]['name'] for _ in dut_route['advertisedTo']])
+            for vm in other_vms:
+                if vm not in advertised_to:
+                    logging.warn("DUT didn't advertise route to neighbor %s" % vm)
```
Comment (wenyiz2021) [inline tests/bgp/test_bgp_bbr.py:252]: Hi @lolyu, what does this check intended to do? 
this doesn't work for multi-asic. 
other_vms by definition includes all t0 and t2, in multi-asic dut these would be on different asics. 
however, 'dut_route' is from tor1_namespace which is a single asic : dut_route = duthost.get_route(route.prefix, setup['tor1_namespace'])

Comment (lolyu) [reply tests/bgp/test_bgp_bbr.py:252]: This is supposed to check the route is advertised to all the neighbors in a single asic SONiC. If it is a multi-asic device, you might need to limit the `other_vms` only to the neighbors in the same asic.

Comment (wenyiz2021) [reply tests/bgp/test_bgp_bbr.py:252]: gotcha

**Pull request merged**